pkg/jobs/registry.go

```diff
@@ -683,6 +683,31 @@ func (r *Registry) getJobFn(ctx context.Context, txn *kv.Txn, id int64) (*Job, R
 	return job, resumer, nil
 }
 
+// CheckJobControlOwnership reads the "owner" of the job referred to by id, and
+// returns true if the user issuing the job control query is the same as the
+// "owner" of the job.
+func (r *Registry) CheckJobControlOwnership(
```

Comment:
this feels out of place to me? I think I'd say the registry should just be able to tell you the owner -- what you _do_ with that should be in the control path, not here.

Combining that with my comment below, if you agree about ignoring the type, then I think it'd just be `if owner, err := reg.Owner(id); ... if !isOwner { p.RequireAdmin..  }`